server/src/controllers/restaurant.ts

```diff
@@ -60,12 +71,23 @@ async function findNearWithinBudget(
     const page: number = parseFloat(request.query.page as string);
     const skip = (!isNaN(page) ? page : 0) * RESTAURANT_LIMIT;
 
+    const priceCategories = request.query.price_categories as PriceCategory[];
+    const foodCategories = request.query.food_categories as FoodCategory[];
+    const transactions = request.query.transactions as TransactionCategory[];
+
+    const restaurantFilter: RestaurantFilter = {
+      price_categories: priceCategories,
+      food_categories: foodCategories,
+      transactions: transactions,
+    };
+
```

Comment:
Lines 34-42 and 74-82 are the same, is this just duped code or am I missing something here?

Reply:
theyre for different routes. One for nearby and another for nearby in budget.

Reply:
But good thing you pointed it out, I can probably make a function that generates the filters from `request.query`

Reply:
ahhh, I see - just github collapsed the code so I didn't realize, my bad should've noticed

Reply:
converted it to function
![image](https://user-images.githubusercontent.com/51217000/235282815-21b1a70b-d22b-400b-8a08-4b7c35a0b794.png)